src/group_theory/torsion.lean

```diff
@@ -188,6 +190,45 @@ lemma torsion.is_torsion : is_torsion $ torsion G :=
     by rw [mul_left_iterate, _root_.mul_one, submonoid.coe_pow,
            subtype.coe_mk, submonoid.coe_one, (is_periodic_pt_mul_iff_pow_eq_one _).mp hn]⟩
 
+variables (G) (p : ℕ) [hp : fact p.prime]
+include hp
+
+/-- The `p`-primary component is the submonoid of elements with order prime-power of `p`. -/
+@[to_additive
+  "The `p`-primary component is the submonoid of elements with additive order prime-power of `p`."]
+def primary_component : submonoid G :=
+{ carrier := {g | ∃ n : ℕ, order_of g = p ^ n},
+  one_mem' := ⟨0, by rw [pow_zero, order_of_one]⟩,
+  mul_mem' := λ g₁ g₂ hg₁ hg₂, exists_order_of_eq_prime_pow_iff.mpr $ begin
+    obtain ⟨m, hm⟩ := exists_order_of_eq_prime_pow_iff.mp hg₁,
+    obtain ⟨n, hn⟩ := exists_order_of_eq_prime_pow_iff.mp hg₂,
+    exact ⟨m + n, by rw [mul_pow, pow_add, pow_mul, hm, one_pow, monoid.one_mul,
+                         mul_comm, pow_mul, hn, one_pow]⟩,
+  end }
+
+variables {G} {p}
+
+/-- Elements of the `p`-primary component have order `p^n` for some `n`. -/
+@[to_additive "Elements of the `p`-primary component have additive order `p^n` for some `n`"]
+lemma primary_component.exists_order_of_eq_prime_pow (g : comm_monoid.primary_component G p) :
+  ∃ n : ℕ, order_of g = p ^ n :=
+by simpa only [comm_monoid.primary_component, subtype.val_eq_coe,
+               set.mem_set_of_eq, submonoid.mem_mk, order_of_submonoid] using g.property
```

Comment:
This is quick enough that I think it's nicer to suppress the noise of the basic lemma names.
```suggestion
by simpa [primary_component] using g.property
```